src/main/java/com/easypost/model/ErrorDeserializer.java

```diff
@@ -12,8 +12,32 @@
 
 import java.lang.reflect.Type;
 import java.util.ArrayList;
+import java.util.Map.Entry;
 
 public final class ErrorDeserializer implements JsonDeserializer<Error> {
+    /**
+     * Recursively traverse an error JSON element and its sub-element(s), and extracts all
+     * error string values found into the specified string list.
+     *
+     * @param element the JSON element to traverse
+     * @param messagesList the list of strings to append found values to
+     */
+    private void traverseJsonElement(JsonElement element, ArrayList<String> messagesList) {
```

Comment:
This works because Java is "pass by reference", so the `messageList` variable getting updated by each recursive call does reflect on the variable on line 65 below.

However, this won't work in other languages that don't pass by reference. Instead, can we make this recursive function actually return the updated/appended `messageList` variable so we don't have to worry about its reference?

Reply:
This will work with the other six languages as they support `pass by reference` too or has `pass by reference` already for passing in an object into a method, some languages might require you to add an additional symbol, like in Go you would need to use a pointer, and in PHP you would need to use `&`.